services/discourse_translator/base.rb

```diff
@@ -31,22 +31,31 @@ def self.access_token
       raise "Not Implemented"
     end
 
-    def self.from_custom_fields(post)
-      post_translated_custom_field =
-        post.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] || {}
-      translated_text = post_translated_custom_field[I18n.locale]
+    def self.from_custom_fields(topic_or_post)
+      translated_custom_field =
+        topic_or_post.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] || {}
+      translated_text = translated_custom_field[I18n.locale]
 
       if translated_text.nil?
         translated_text = yield
 
-        post.custom_fields[
+        topic_or_post.custom_fields[
           DiscourseTranslator::TRANSLATED_CUSTOM_FIELD
-        ] = post_translated_custom_field.merge(I18n.locale => translated_text)
+        ] = translated_custom_field.merge(I18n.locale => translated_text)
 
-        post.save!
+        topic_or_post.save!
       end
 
       translated_text
     end
+
+    def self.get_text(topic_or_post)
+      case topic_or_post.class.name
+      when "Post"
+        text = topic_or_post.cooked
+      when "Topic"
+        topic_or_post.title
+      end
+    end
```

Comment:
Hmm. I think here is where it gets grey.

In the subclasses it won't be obvious we're getting a topic title text. I'm wondering if it's worth it to think if the base class needs a better interface?